classes/log/store.php

```diff
@@ -167,6 +167,16 @@ public function get_max_batch_size() {
         return $this->get_config('maxbatchsize', 100);
     }
 
+    private function get_successful_events($events) {
```

Comment:
Has this been copied exactly from the emit task?
If so, should we move these functions out to the shared lib?

Reply:
I'm always worried about cron tasks pulling in too much from lib.php
Being a private member function minimises the dependencies on lib.php
Would it be better to include this in lib.php?

Reply:
I'd be keen to hear what the others think on this.
I'm thinking about duplication of code point of view, but can't comment from a moodle cron perspective

Reply:
I'm thinking we should avoid duplication of code to be honest. Perhaps we could use inheritance within these classes to avoid including the lib file?

Reply:
I don't like private function because it can not be inherited.
I don't think this function is necessary here.
I think All of these changes should be set in the emit task.

Reply:
This one is in the lib file now, isn't it?

Reply:
Yes it is in lib.php

Reply:
This function should be removed from `classes/log/store.php`